### PR TITLE
kube zombie handling

### PR DESCRIPTION
**Pull request by jrhizor**
Kills pod with an error code if he caller isn't available. Also adds integration tests for the pod process and handles some additional edge cases for waiting for pod statuses and pipes in entrypoints.

### CODE REVIEW

File: kube/resources/scheduler.yaml [hunk 1]
```diff
@@ -87,8 +87,8 @@ spec:
                   name: airbyte-env
                   key: TEMPORAL_WORKER_PORTS
           ports:
-            - containerPort: 9000
-            - containerPort: 9001
+            - containerPort: 9000 # for heartbeat server
```
Comment (davinchia) [inline kube/resources/scheduler.yaml:90]: nice!

File: airbyte-scheduler/app/src/main/java/io/airbyte/scheduler/app/SchedulerApp.java [hunk 2]
```diff
@@ -209,6 +211,19 @@ public static void main(String[] args) throws IOException, InterruptedException
         jobPersistence);
     final JobNotifier jobNotifier = new JobNotifier(configs.getWebappUrl(), configRepository);
 
+    if (configs.getWorkerEnvironment() == Configs.WorkerEnvironment.KUBERNETES) {
+      Map<String, String> mdc = MDC.getCopyOfContextMap();
+      Executors.newSingleThreadExecutor().submit(
+              () -> {
+                MDC.setContextMap(mdc);
```
Comment (davinchia) [inline airbyte-scheduler/app/src/main/java/io/airbyte/scheduler/app/SchedulerApp.java:218]: is this so

File: airbyte-scheduler/app/src/main/java/io/airbyte/scheduler/app/kube/WorkerHeartbeatServer.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Map;
+
+public class WorkerHeartbeatServer {
```
Comment (davinchia) [inline airbyte-scheduler/app/src/main/java/io/airbyte/scheduler/app/kube/WorkerHeartbeatServer.java:18]: nit: comments here to capture what this is doing.

File: airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java [hunk 4]
```diff
@@ -188,19 +198,23 @@ private static Container getInit(boolean usesStdin, List<VolumeMount> mainVolume
         .build();
```
Comment (davinchia) [inline airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java:198]: we should add the heartbeating description to this class' doc string.

**Review by davinchia — APPROVED** [review]
Generally looks good. Tests aside, I'd want us to have documentation on the KubePodProcess level (this is how the process heart beats) and the Worker level (this is how we expect heart beats to work at the job level). No preference on where the documentation goes, except it be centralised and the various start up scripts should also point to it so someone chancing upon those scripts know where to go for understanding.

File: airbyte-scheduler/app/src/main/java/io/airbyte/scheduler/app/SchedulerApp.java (hunk 2 above)
Comment (jrhizor) [reply airbyte-scheduler/app/src/main/java/io/airbyte/scheduler/app/SchedulerApp.java:218]: We want to see logs if this fails.

File: airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java [hunk 5]
```diff
@@ -346,7 +389,10 @@ public KubePodProcess(KubernetesClient client,
     copyFilesToKubeConfigVolume(client, podName, namespace, filesWithSuccess);
 
     LOGGER.info("Waiting until pod is ready...");
-    client.resource(podDefinition).waitUntilReady(30, TimeUnit.MINUTES);
+    client.resource(podDefinition).waitUntilCondition(p -> {
+      boolean isReady = Objects.nonNull(p) && Readiness.getInstance().isReady(p);
+      return isReady || isTerminal(p); // todo: fixes too fast completion, but what do we do for failure/error?
```
Comment (davinchia) [inline airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java:394]: add clarifying comment about possible error states, even though we don't think this is possible

**Pull request merged**